### PR TITLE
Update min/max version test to include new IndexVersions

### PR DESCRIPTION
**Pull request by thecoop**
Add tests for the new index version methods on `DiscoveryNodes`, that it calculates the min/max versions correctly

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java [hunk 1]
```diff
@@ -401,48 +402,42 @@ Set<String> matchingNodeIds(DiscoveryNodes nodes) {
...
-            )
-        );
         DiscoveryNodes build = discoBuilder.build();
-        assertEquals(Version.fromString("6.3.0"), build.getMaxNodeVersion());
-        assertEquals(Version.fromString("1.1.0"), build.getMinNodeVersion());
+
+        assertEquals(Version.fromString("5.0.17"), build.getMaxNodeVersion());
+        assertEquals(Version.fromString("1.6.0"), build.getMinNodeVersion());
+        assertEquals(Version.fromString("2.1.0"), build.getSmallestNonClientNodeVersion());
+        assertEquals(IndexVersion.fromId(2010099), build.getMaxDataNodeCompatibleIndexVersion());
```
Comment (williamrandolph) [inline server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java:439]: This ID loosely translates to "2.0.0", right? Why is it the max data node compatible index version? Is it basically a way of saying "2.x"?

Comment (thecoop) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java:439]: This is the lowest max index version of all nodes that can store data

**Review by rjernst — APPROVED** [review]
LGTM, just a couple very minor asks

File: server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java [hunk 2]
```diff
@@ -401,48 +402,42 @@ Set<String> matchingNodeIds(DiscoveryNodes nodes) {
...
-            new DiscoveryNode(
-                "name_" + 1,
-                "node_" + 1,
-                buildNewFakeTransportAddress(),
-                Collections.emptyMap(),
-                new HashSet<>(randomSubsetOf(DiscoveryNodeRole.roles())),
-                Version.fromString("5.1.0")
-            )
+        List<VersionInformation> dataVersions = List.of(
+            new VersionInformation(Version.fromString("3.2.5"), IndexVersion.fromId(2000099), IndexVersion.fromId(3020599)),
```
[the remark this answers is not in the document]
Comment (thecoop) [reply server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java:413]: I've added some comments on which nodes are ignored for each result

**Pull request merged**